### PR TITLE
Fix/fix outstanding rx bloc cli issues

### LINKED ISSUE #663: Stabilisation improvements
As a developer I want the projects generated with rx_bloc_cli to be as stable as possible.

ACs

- [x] Build errors with `enable-analytics` flag enabled
- [x] CD compatibilities issues with Flutter 3.19
- [x] Project path default value in the Github Workflow files (part of the project CD)
- [x] CD Configuration documentation/Fastfile:
    - [x] Update documentation for setting up Apple Team ID
    - [x] Install all provisioning profiles (un-comment existing provisioning profiles)
    - [x] gym paths
    - [x] update credentials and fastfile to use a consistent naming scheme (ex: snake_case )
    - [x] install only needed provisioning profile from provisioning_profile_file_name_map dictionary
    - [x] provisioning_profile_file_name_map (file names with extensions) documentation
- [x] Golden tests generated with the `enable-feature-counter` flag enabled
- [x] Generated shell script running issues
- [x] Add documentation for updating project to manual signing of project and selecting the right provisioning profiles
- [x] `ios/Podfile`: configure ios deployment target (min sdk version `12.0`)

### PR DESCRIPTION
**Pull request by DDavidPrime**
Closes #663

### CODE REVIEW

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md [hunk 1]
```diff
@@ -23,7 +23,24 @@ export MOBILE_DISTRIBUTION_ENCRYPTION_PASSWORD=<PASSWORD>
...
-### Fastfile amendments
+### Project & Fastfile amendments
+
+For the iOS project, make sure you set project signing to manual and apply expected provisioning profile for each build configuration.
+To do that, you need to:
+- Open project in XCode
+- Inside the targets section, select your `Runner` target
+- Open the `Signing & Capabilities` tab
+- Uncheck the `Automatically manage signing` checkmark in each of the build configurations
+- Select a fitting provisioning profile for each of the build configurations
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/docs/continuous_delivery.md:34]: This is not true. Only for the release  configurations the automatic signing needs to be turned off

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md [hunk 2]
```diff
@@ -23,7 +23,24 @@ export MOBILE_DISTRIBUTION_ENCRYPTION_PASSWORD=<PASSWORD>
...
+- Save the changes in XCode
+
+Inside the `{project_root}/bin/` directory, you'll find several scripts generated by the rx_bloc_cli package.
+Among others, make sure to find the `build_runner_build.sh` which is used during CI/CD pipeline. 
+Change the execution privileges of that file using the following command:
+
+```
+sudo chmod 755 path/to/file
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/docs/continuous_delivery.md:42]: Why we need to use sudo ? Can't we execute this automatically from the rx_bloc_cli crate ? chmod +x

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md [hunk 3]
```diff
@@ -35,18 +52,31 @@ Take that value and paste it into the `IOS_P8_AUTH_KEY_ID` variable.
 The value of the `IOS_P8_AUTH_KEY_ISSUER_ID` variable is the value retrieved from the `contentProviderPublicId` key within [this page][apple_issuer_id_details].
 If more than one `contentProviderPublicId` with different values is present, make sure to use the one from the appropriate organization account name.
 
+Update the `PROJECT_DIRECTORY_NAME` variable with the name of the root directory in which the project is located.
+
+Copy the url of the distribution repository pa paste it inside the `DISTRIBUTION_REPOSITORY_URL` variable.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/docs/continuous_delivery.md:57]: What pa means ?

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md [hunk 4]
```diff
@@ -35,18 +52,31 @@ Take that value and paste it into the `IOS_P8_AUTH_KEY_ID` variable.
 The value of the `IOS_P8_AUTH_KEY_ISSUER_ID` variable is the value retrieved from the `contentProviderPublicId` key within [this page][apple_issuer_id_details].
 If more than one `contentProviderPublicId` with different values is present, make sure to use the one from the appropriate organization account name.
 
+Update the `PROJECT_DIRECTORY_NAME` variable with the name of the root directory in which the project is located.
+
+Copy the url of the distribution repository pa paste it inside the `DISTRIBUTION_REPOSITORY_URL` variable. 
+Make sure you include the url without the `http://` and `https://` prefixes, as this path will be combined with the distribution repository access token.
```
Comment (StanevPrime) [inline packages/rx_bloc_cli/example/docs/continuous_delivery.md:58]: Provide an example

**Review by StanevPrime — CHANGES_REQUESTED** [review]

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md (hunk 3 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/example/docs/continuous_delivery.md:57]: Typo, should say `and`.

File: packages/rx_bloc_cli/example/docs/continuous_delivery.md (hunk 2 above)
Comment (DDavidPrime) [reply packages/rx_bloc_cli/example/docs/continuous_delivery.md]: Should be addressed through code generation now...

**Review by StanevPrime — APPROVED** [review]

**Pull request merged**